codec/proto_codec.go

```diff
@@ -11,13 +11,21 @@ import (
 	"github.com/gogo/protobuf/proto"
 )
 
+// ProtoCodecMarshaler defines and interface for codecs that utilize Protobuf for both
+// binary and JSON encoding.
+type ProtoCodecMarshaler interface {
+	Marshaler
+	InterfaceRegistry() types.InterfaceRegistry
```

Comment:
I don't think we want that - this is used for Amino.

Reply:
What is used for amino?